### PR TITLE
Db

### PR DESCRIPTION
**Pull request by entopia**
Added tables for all other datasets

### CODE REVIEW

File: crop/structure.py [hunk 1]
```diff
@@ -143,19 +175,125 @@ class ReadingsTinyTagsClass(BASE):
...
+    __tablename__ = CROP_SHEET_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+    crop = Column(String, nullable=False)
+    propagation_date = Column(DateTime, nullable=False)
+    column = Column(Integer, nullable=False)
+    aisle = Column(String, nullable=False)
+    shelf = Column(Integer, nullable=False)
```
Comment (tomaslaz) [inline crop/structure.py]: Is this location? Do you think we could join with the locations table instead?

File: crop/structure.py [hunk 2]
```diff
@@ -143,19 +175,125 @@ class ReadingsTinyTagsClass(BASE):
...
+    yield_m2 = Column(Integer, nullable=False)
+    propagation_days = Column(Integer, nullable=False)
+    days_under_lights = Column(Integer, nullable=False)
+    unique_code = Column(String, nullable=False)
+    surplus_wasted = Column(Integer, nullable=False)
+    x_g = Column(Integer, nullable=False)
+    total_waste_g = Column(Integer, nullable=False)
+    projected_yeild_m = Column(Integer, nullable=False)
+    projected_yield_tot = Column(Integer, nullable=False)
+    num_trays_wasted = Column(String, nullable=True)
```
Comment (tomaslaz) [inline crop/structure.py]: If it is a number, perhaps it should be Float?

File: crop/structure.py [hunk 3]
```diff
@@ -143,19 +175,125 @@ class ReadingsTinyTagsClass(BASE):
...
+    waste_explanation = Column(String, nullable=False) #no ref on what that is
+    yield_m2 = Column(Integer, nullable=False)
+    propagation_days = Column(Integer, nullable=False)
+    days_under_lights = Column(Integer, nullable=False)
+    unique_code = Column(String, nullable=False)
+    surplus_wasted = Column(Integer, nullable=False)
+    x_g = Column(Integer, nullable=False)
+    total_waste_g = Column(Integer, nullable=False)
+    projected_yeild_m = Column(Integer, nullable=False)
+    projected_yield_tot = Column(Integer, nullable=False)
```
Comment (tomaslaz) [inline crop/structure.py]: If it is a number, perhaps it should be Float?

File: crop/structure.py [hunk 4]
```diff
@@ -143,19 +175,125 @@ class ReadingsTinyTagsClass(BASE):
...
+    __tablename__ = OTHER_SHEET_TABLE_NAME
+
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    tank_time = Column(DateTime, nullable=False)
+    tank_no = Column(Integer, nullable=False)
+    tank_ph = Column(Float, nullable=False)
+    tank_ec = Column(Float)
+    tank_clox = Column(Float)
+    tank_water_temp = Column(Integer)
```
Comment (tomaslaz) [inline crop/structure.py]: If it is a number, perhaps it should be Float?

Comment (tomaslaz) [inline crop/structure.py]: nullable=False ?

File: crop/constants.py [hunk 5]
```diff
@@ -111,5 +111,10 @@
 LOCATION_TABLE_NAME = "locations"
 ADVANTIX_READINGS_TABLE_NAME = "advantix_data"
 TINYTAGS_READINGS_TABLE_NAME = "microtag_data"
+AIR_VELOCITY_READINGS_TABLE_NAME = "air_velocity_data"
+NEW_SENSOR_TABLE_NAME = "melanie_data"
```
[the remark this answers is not in the document]
Comment (misspawty) [reply crop/constants.py]: In the documentation, these are actually mentioned as "air velocity sensors". They report to ras 6 and 7. 

The new sensors are also mentioned as "New environmental sensors". THese are the custom made ones by melanie. dont have a name .

File: crop/constants.py [hunk 6]
```diff
@@ -111,5 +111,10 @@
 LOCATION_TABLE_NAME = "locations"
 ADVANTIX_READINGS_TABLE_NAME = "advantix_data"
 TINYTAGS_READINGS_TABLE_NAME = "microtag_data"
+AIR_VELOCITY_READINGS_TABLE_NAME = "air_velocity_data"
```
[the remark this answers is not in the document]
Comment (misspawty) [reply crop/constants.py]: op

File: crop/constants.py [hunk 7]
```diff
@@ -111,5 +111,10 @@
...
+NEW_SENSOR_TABLE_NAME = "melanie_data"
+ENERGY_READINGS_TABLE_NAME = "energy_data"
```
[the remark this answers is not in the document]
Comment (misspawty) [reply crop/constants.py]: Energy readings are from the point meters. not from sensors

File: crop/structure.py [hunk 8]
```diff
@@ -10,6 +10,8 @@
     ForeignKey,
     Column,
     Integer,
+    Float,
+    DECIMAL,
```
[the remark this answers is not in the document]
Comment (misspawty) [reply crop/structure.py]: I read that decimal is good for use in percentages. I suppose it is the same, could use one or the other. 
I used decimal with 2,5 to signify that this is a percentage column. and add a restriction on 0-1 (to do).

File: crop/structure.py [hunk 9]
```diff
@@ -123,12 +133,34 @@ class ReadingsAdvantixClass(BASE):
     time_updated = Column(DateTime(), onupdate=func.now())
 
 
-class ReadingsTinyTagsClass(BASE):
+class ReadingsAirVelocity(BASE):
```
[the remark this answers is not in the document]
Comment (misspawty) [reply crop/structure.py]: yeap. will add class

File: crop/structure.py [hunk 10]
```diff
@@ -123,12 +133,34 @@ class ReadingsAdvantixClass(BASE):
...
     """
-    Class for reading the raw Advantix data
+    Base class for the raw Air Velocity data readings
     """
 
-    __tablename__ = TINYTAGS_READINGS_TABLE_NAME
+    __tablename__ = AIR_VELOCITY_READINGS_TABLE_NAME
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
```
[the remark this answers is not in the document]
Comment (misspawty) [reply crop/structure.py]: :)

File: crop/structure.py (hunk 1 above)
Comment (misspawty) [reply crop/structure.py]: yes! that was my next step. pretty nice join.

File: crop/structure.py [hunk 11]
```diff
@@ -143,19 +175,125 @@ class ReadingsTinyTagsClass(BASE):
...
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+    crop = Column(String, nullable=False)
+    propagation_date = Column(DateTime, nullable=False)
+    column = Column(Integer, nullable=False)
+    aisle = Column(String, nullable=False)
+    shelf = Column(Integer, nullable=False)
+    trays = Column(Integer, nullable=False)
+    m2 = Column(Integer, nullable=False)
```
[the remark this answers is not in the document]
Comment (misspawty) [reply crop/structure.py]: this one is the number of trays, which is usually an integer. 
The m2 you are right i ll change.

File: crop/structure.py (hunk 2 above)
Comment (misspawty) [reply crop/structure.py]: Now that i have checked, they seem to be using integers for all these columns.  eg. 2659.

File: crop/structure.py (hunk 3 above)
Comment (misspawty) [reply crop/structure.py]: so trays are definitely integer number as they refer to number of trays. and propagation days etc. The m2, g, and tot can possibly convert to floats, but they seem to be using always integers in their reports.

File: crop/structure.py (hunk 4 above)
Comment (misspawty) [reply crop/structure.py]: same, ph seems to be float numbers, water temp seems to be measured and reported as integer

Comment (misspawty) [reply crop/structure.py]: These have lots of null values. Don know if it is intentional, or missing, but it is possible that some times these manual values may not be reported or need to stay null?

File: crop/constants.py [hunk 12]
```diff
@@ -111,5 +111,10 @@
...
+CROP_SHEET_TABLE_NAME = "crop_data"
+OTHER_SHEET_TABLE_NAME = "other_data"
```
[the remark this answers is not in the document]
Comment (entopia) [reply crop/constants.py]: crop growth data
infrastructure data

File: crop/structure.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (entopia) [reply crop/structure.py]: convert all decimals to float

File: crop/structure.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (entopia) [reply crop/structure.py]: change to float

File: crop/structure.py (hunk 2 above)
Comment (entopia) [reply crop/structure.py]: change to floats

**Comment by entopia** [comment]
This is a working version of the Advanticsys dataset . 
The test have been really useful! I am starting to understand the value of test oriented development.  I would have been lost without that structure of the code. 
Tests pass in my system, not quite sure why Travis fails. Would it be possible to check if it works for you Tomas?

**Pull request closed without merging**